pkg/transport/proxy/transparent/transparent_proxy.go

```diff
@@ -66,15 +83,115 @@ func NewTransparentProxy(
 		middlewares:       middlewares,
 		shutdownCh:        make(chan struct{}),
 		prometheusHandler: prometheusHandler,
+		sessions:          make(map[string]session.Session),
 	}
 
 	// Create MCP pinger and health checker
 	mcpPinger := NewMCPPinger(targetURI)
 	proxy.healthChecker = healthcheck.NewHealthChecker("sse", mcpPinger)
 
+	_, err := proxy.CreateSession(TransparentProxySessionID)
+	if err != nil {
+		logger.Errorf("Failed to create session for TransparentProxy: %v", err)
+		return nil
+	}
+
 	return proxy
 }
 
+func (p *TransparentProxy) handleModifyResponse(res *http.Response) error {
+	// Log headers
+	if sid := res.Header.Get("Mcp-Session-Id"); sid != "" {
+		logger.Infof("🆔 Streamable session ID from header: %s", sid)
+	}
+
+	// Handle streaming (SSE)
+	if ct, _, _ := mime.ParseMediaType(res.Header.Get("Content-Type")); ct == "text/event-stream" {
+		pr, pw := io.Pipe()
+		orig := res.Body
+		res.Body = pr
+
+		go func() {
+			defer pw.Close()
+			scanner := bufio.NewScanner(orig)
+			re := regexp.MustCompile(`sessionId=([\w-]+)`) // Capture UUID-like IDs
+
+			for scanner.Scan() {
+				line := scanner.Text()
+
+				if matches := re.FindStringSubmatch(line); len(matches) == 2 {
+					sessionID := matches[1]
+
+					// set session id for proxy
+					extractedSession, ok := p.GetSession(TransparentProxySessionID)
+					if !ok {
+						logger.Errorf("Failed to get session for TransparentProxy")
+						continue
+					}
+					extractedSession.SetIsInitialized(true)
+					extractedSession.SetMCPSessionID(sessionID)
+				}
+				_, err := pw.Write([]byte(line + "\n"))
+				if err != nil {
+					logger.Errorf("Failed to write to pipe: %v", err)
+				}
+			}
+		}()
+		return nil
+	}
+
+	// Handle non-streaming (JSON) bodies
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+	err = res.Body.Close()
+	if err != nil {
+		logger.Errorf("Failed to close response body: %v", err)
+	}
+
+	text := string(body)
+	logger.Infof("HTTP response body: %s", text)
+
+	// Parse sessionId if embedded in JSON
+	rejson := regexp.MustCompile(`"sessionId"\s*:\s*"([\w-]+)"`)
+	if matches := rejson.FindStringSubmatch(text); len(matches) == 2 {
+		sessionID := matches[1]
+		logger.Infof("🆔 Captured sessionId from JSON: %s", sessionID)
```

Comment:
Same comment regarding regex - this may mess up some places which do not expect unicode characters.